File: Second Largest - GFG/second-largest.java
//{ Driver Code Starts
//Initial Template for Java




import java.util.*;
import java.io.*;

public class Main {

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int tc = Integer.parseInt(br.readLine().trim());
        while (tc-- > 0) {
            String[] inputLine;
            int n = Integer.parseInt(br.readLine().trim());
            int[] arr = new int[n];
            inputLine = br.readLine().trim().split(" ");
            for (int i = 0; i < n; i++) {
                arr[i] = Integer.parseInt(inputLine[i]);
            }

            int ans = new Solution().print2largest(arr, n);
            System.out.println(ans);
        }
    }
}

// } Driver Code Ends


//User function Template for Java

class Solution {
    int print2largest(int arr[], int n) {
        int lar= arr[0], slar = -1;
        for(int i = 0;i < n; i++) {
            if(arr[i] > lar) {
                slar = lar;
                lar = arr[i];
            }
            if(arr[i] > slar && arr[i] < lar){
                slar = arr[i];
            }
        }
        return slar;
    }
}

File: Second Largest - GFG/second-largest.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>

using namespace std;

// } Driver Code Ends
//User function template for C++
class Solution{
public:	
	// Function returns the second
	// largest elements
	int print2largest(int arr[], int n) {
	   int lar = arr[0];
	   int slar = -1;
	   for(int i = 0; i < n; i++) {
	       if(lar < arr[i]) {
	           slar = lar;
	           lar = arr[i];
	       }
	       if(lar != arr[i] && slar < arr[i]) {
	           slar = arr[i];
	       }
	   }
	   return slar;
	}
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        int arr[n];
        for (int i = 0; i < n; i++) {
            cin >> arr[i];
        }
        Solution ob;
        auto ans = ob.print2largest(arr, n);
        cout << ans << "\n";
    }
    return 0;
}

// } Driver Code Ends